deep-learning/src/test/python/synapsemltest/dl/conftest.py

```diff
@@ -82,3 +92,22 @@ def transform():
         ]
     )
     return transform
+
+
+def _prepare_text_data(spark):
+    if not os.path.exists(join(dataset_dir, "target/Emotion_classification.csv")):
+        urllib.request.urlretrieve(
+            "https://mmlspark.blob.core.windows.net/publicwasb/text_classification/Emotion_classification.csv",
+            dataset_dir + "target/Emotion_classification.csv",
+        )
+    df = pd.read_csv(dataset_dir + "target/Emotion_classification.csv")
+    df = spark.createDataFrame(df)
+    indexer = StringIndexer(inputCol="Emotion", outputCol="label")
+    indexer_model = indexer.fit(df)
+    df = indexer_model.transform(df).drop("Emotion")
+
+    train_df, test_df = df.randomSplit([0.85, 0.15], seed=1)
+
+    os.remove(dataset_dir + "target/Emotion_classification.csv")
```

Comment:
reading using spark will allow you to avoid mutable state that you need to clean up

Reply:
Thx!